cmd/dagger/call.go

```diff
@@ -9,6 +9,50 @@ import (
 	"github.com/spf13/cobra"
 )
 
+// TODO: remove these ghost commands after next release (v0.9.8)
+var downloadCmd = &cobra.Command{
+	Use:                "download",
+	Short:              "Download an asset from module function",
+	Aliases:            []string{"export", "dl"},
+	Hidden:             true,
+	SilenceUsage:       true,
+	DisableFlagParsing: true,
+	Args: func(cmd *cobra.Command, args []string) error {
+		return fmt.Errorf(`%q has been replaced by "dagger call COMMANDS... --output=PATH"`, cmd.CommandPath())
```

Comment:
Would have expected to see `export` after `COMMANDS...` here - do we special-case this?

Reply:
Yeah, we could. I didn't because the point of `--output` is to provide more symmetry in the boundary between host and dagger. We had input flags, but no equivalent output concept. @sipsma wrote about this in [`-o` flag for writing to caller's filesystem](https://github.com/dagger/dagger/issues/6229#issuecomment-1863649336)

> * E.g. you can specify an input `Directory` arg as `--dir ./some/input/dir` but to _write_ a `Directory` to the caller's filesystem you have to go through the whole `... export --path ./some/output/dir`
> * This is okay generally speaking, but does create some asymmetry
> 
> Coming out of that, @shykes brought up idea of supporting a generic `-o <path>` (`--output <path>` in long form) flag on the CLI command. This flag would be applicable for any return type and result in that result being written to the caller's filesystem rather than written to stdout.

As you can see `dagger call build -o bin/` is shorter than `dagger call build export --path=bin/`, and it also works with most other types:

```console
# dagger call test stdout -o test.$(date +%s).log
```

However, since these "ghost" commands are just to make transition easier for next release, where many commands will fail to run, I'm not against making this consistent with the others and also choosing core chaining to `export` explicitly.

Reply:
I'm good with the shorthand given how common it'll probably be, just didn't know about it. :+1: IMO we should point to it if that's what we expect folks to do instead of the explicit `export`